src/ServiceControl/MessageFailures/Handlers/UnArchiveMessagesHandler.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace ServiceControl.MessageFailures.Handlers
+{
+    using System.Linq;
+    using Contracts.MessageFailures;
+    using InternalMessages;
+    using NServiceBus;
+    using Raven.Client;
+
+    public class UnArchiveMessagesHandler : IHandleMessages<UnArchiveMessages>
+    {
+        public IDocumentSession Session { get; set; }
```

Comment:
OK so the outgoing MSMQ message (representing the event) will be rolled back if the Unit of Work fails? Do all of the transports capture outgoing messages in the receive transaction?